### ISSUE TITLE
`rulesDirectory` in `tslint.json` doesn't work in tslint 3.5.0

### ISSUE THREAD

**Issue by aleung** [palantir/tslint#1007]
After upgrade tslint from 3.2 to 3.5.0, the `rulesDirectory` in `tslint.json` doesn't work. It reports error " Could not find the following rules specified in the configuration".

I added a log to print `rulesDirectories` passing into `loadRules()` and found it's null.

**Comment by aleung** [comment]
This issue exists in 3.4.0 and 3.5.0.
Rollback to 3.3.0 it works.

**Comment by JKillian** [comment]
Hi @aleung, I believe the difference you're seeing is that TSLint since version 3.4.0 warns you about rules it couldn't find whereas 3.3.0 just failed silently. Can I ask how you're running TSLint? Is it through the command-line, through grunt-tslint, through tslint-loader, etc?

**Comment by jvanoostveen** [comment]
We are seeing this message when using WebStorm and Visual Studio Code.
Can't read the entire error btw, but I guess that issue is WebStorm related (not enough room for the error).

**Comment by JKillian** [comment]
I think what's going on is that these external tools aren't passing the `rulesDirectory` option from `tslint.json` to TSLint correctly. This is sort of a subtle bug, so it's not a surprise that there's issues with this. I explained what I've seen happening in [tslint-loader here](https://github.com/wbuchwalter/tslint-loader/issues/16). I'll think about ways this can be improved on the end of TSLint.

All this being said, I'm 99% confident that TSLint 3.4.0 and later did _not_ introduce a new bug regarding this, these versions are simply more vocal when a rule cannot be found. I believe that in TSLint 3.3.0 things are just failing silently - the rules that can't be found are just being ignored and are not getting run on your code.

**Comment by aleung** [comment]
@JKillian I'm using gulp-tslint 4.3.3

**Comment by JKillian** [comment]
@aleung As a temporary workaround, try also specifying the paths to custom rules directories in your gulp file, like such:

```js
.pipe(tslint({
  rulesDirectory: "path/to/rules/"
}))
```

**Comment by JKillian** [comment]
Related: https://github.com/panuhorsmalahti/gulp-tslint/issues/55

**Comment by art0rz** [comment]
I'm also seeing this error. I'm using `tslint` directly from CLI, with `rulesDirectory` set in my `tslint.config`. I've used the no-imports example from the documentation. TSLint throws an error about not being able to find the `no-imports` rule source. When I run the cli command with `--rules-dir` it does work.

my config: 
```json
{
  "rulesDirectory:": "./tslint/",
  "rules": {
    "no-imports": true,

    // ...
```

I'm running tslint `3.11.0`

**Comment by JKillian** [comment]
The paths specified in config files are relative to the location of the config file, where as the paths specified via the CLI are relative to the CWD. Does this relate to the issue you're having?

If not, can you push a minimal repo up to github with your setup?

**Comment by art0rz** [comment]
@JKillian you're right. It does work with a bare-bones project I just set up. Must have been something else in the project's environment. Thanks!

**Comment by JKillian** [comment]
Sweet, I'm going to close this issue then because I think all issues should've been resolved relating to this

**Comment by karfau** [comment]
For everybody thinking about copying from @art0rz snippet that didn't work for him,
there is an additional `:` at the end of the key, removing it makes it work. :)
